compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -3500,6 +3503,105 @@ public void visit(BLangInvocation.BLangActionInvocation aInv, AnalyzerData data)
         checkActionInvocation(aInv, varRef.getBType(), data);
     }
 
+    @Override
+    public void visit(BLangInvocation.BLangResourceAccessInvocation resourceAccessAct, AnalyzerData data) {
+        // Find the lhs expression type
+        checkExpr(resourceAccessAct.expr, data);
+        BType lhsExprType = resourceAccessAct.expr.getBType();
+        
+        if (lhsExprType.tag != TypeTags.OBJECT || !Symbols.isFlagOn(lhsExprType.tsymbol.flags, Flags.CLIENT)) {
+            dlog.error(resourceAccessAct.expr.pos, 
+                    DiagnosticErrorCode.CLIENT_RESOURCE_ACCESS_ACTION_IS_ONLY_ALLOWED_ON_CLIENT_OBJECTS);
+            data.resultType = symTable.semanticError;
+            return;
+        }
+        
+        BObjectTypeSymbol objectTypeSym = (BObjectTypeSymbol) lhsExprType.tsymbol;
+
+        validateResourceAccessPathSegmentTypes(resourceAccessAct.resourceAccessPathSegments, data);
+                
+        // Filter all the resource methods defined on target resource path
+        List<BResourceFunction> resourceFunctions = new ArrayList<>();
+        data.isResourceAccessPathSegments = true;
+        for (BAttachedFunction targetFunc : objectTypeSym.attachedFuncs) {
+            if (Symbols.isResource(targetFunc.symbol)) {
+                BResourceFunction resourceFunction = (BResourceFunction) targetFunc;
+                BLangExpression clonedResourceAccPathSeg = 
+                        nodeCloner.cloneNode(resourceAccessAct.resourceAccessPathSegments);
+                BType resolvedType = checkExprSilent(clonedResourceAccPathSeg, resourceFunction.resourcePathType, data);
+                if (resolvedType != symTable.semanticError) {
+                    resourceFunctions.add(resourceFunction);
+                }
+            }
+        }
+        
+        if (resourceFunctions.size() == 0) {
+            dlog.error(resourceAccessAct.resourceAccessPathSegments.pos, DiagnosticErrorCode.UNDEFINED_RESOURCE);
+            data.resultType = symTable.semanticError;
+            return;
+        }
+        
+        // Filter the resource methods in the list by resource access method name
+        resourceFunctions.removeIf(func -> !func.accessor.value.equals(resourceAccessAct.name.value));
+        
+        if (resourceFunctions.size() == 0) {
+            dlog.error(resourceAccessAct.pos, DiagnosticErrorCode.UNDEFINED_RESOURCE_METHOD, resourceAccessAct.name);
+            data.resultType = symTable.semanticError;
+            return;
+        } else if (resourceFunctions.size() > 1) {
+            dlog.error(resourceAccessAct.pos, DiagnosticErrorCode.AMBIGUOUS_RESOURCE_ACCESS_NOT_YET_SUPPORTED);
+            data.resultType = symTable.semanticError;
+            return;
+        }
+
+        BResourceFunction targetResourceFunc = resourceFunctions.get(0);
```

Comment:
Is this part reachable because of the above if - else if check?

Reply:
Yes we reach here only if `resourceFunctions.size()` is `1`